README.md

```diff
@@ -277,15 +278,46 @@ will be interpeted or passed along.
 All single word arguments (`-j4`, `--impure` etc) 
 are passed to the underlying nix invocation.
 
-##### Tracked files
+#### Tracked files
+
+`nix-direnv` makes a performance tradeoff 
+and only considers changes in a limited number of files 
+when deciding to update its cache.
+
+- for `use nix` this is:
+    * `~/.direnvrc`
+    * `~/.config/direnv/direnvrc`
+    * `.envrc`, 
+    * A single nix file. In order of preference:
+        + The file argument to `use nix`
+        + `default.nix` if it exists
+        + `shell.nix` if it exists
+
+- for `use flake` this is:
+    * `~/.direnvrc`
+    * `~/.config/direnv/direnvrc`
+    * `.envrc`
+    * `flake.nix`
+    * `flake.lock`
+    * `devshell.toml` if it exists
+
+To add more files to be checked use `nix_direnv_watch_file` like this
+
+```shell
+nix_direnv_watch_file your-file.nix
+use nix # or use flake
+```
 
-`nix-direnv` makes a performance tradeoff only considers changes in a limited
-number of files when deciding to update its cache.
+Or - if you don't mind the overhead (runtime and conceptual) of watching all nix-files:
+
+```shell
+nix_direnv_watch_file $(find . -name "*.nix" -printf '"%p"')
```

Comment:
This is not right either. There is no space in between filenames.

```
while IFS= read -r -d $'\0' file; do nix_direnv_watch_file "$file"; done < <(find . -name "*.nix" -type f -print0)
```

Reply:
I actually found that:

```
while IFS= read -r -d $'\0' file; do
  nix_watches+=("$file");
done < <(find . -name "*.nix" -type f -print0)
watch_file "${nix_watches[@]}"
```

Is way faster... maybe we should make a `nix_direnv_watch_dir` instead?

Reply:
Good find. I wrote this too quickly and didn't properly review. Sorry for that.

The correct invocation should be `nix_direnv_watch_file $(find . -name "*.nix" -printf '"%p" ')` (note the trailing space).

I am wholly unsurprised that a solution that avoids calling `[ -e ...]` for every file is faster.
However, I suppose that I dont know if it matters unless the user has huge amounts of nix files.
I have found it to be useful (in this changeset) to have the existence checks in one local spot so that we only ever add extant files to the watch list.

Generally, I think we should be watching *every* nix file. I recognize that that's overzealous, but it is the way to avoid a pretty nasty problem that leads to quite a bit of confusion on the user's behalf. Re-evaluating the devShell too frequently is generally better than not doing it enough.

I'm fine with adding `nix_direnv_watch_dir` taking two arguments: the directory to watch and the extension or glob to find (basically the parameters to invoke `find`).

Reply:
See https://github.com/nix-community/nix-direnv/pull/225 for some micro optimization (maybe like 6% speed up for flake projects)

Reply:
In my dotfiles I have 271 nix files and I can feel it's a slower if I add the slower version vs. the optimized one.

Reply:
okay - that's a lot of nix files...
Significantly more than I had expected a single project to reasonably have :P

How much is the speed up?
Does that 6% hold up here? Or is it greater for more nix files?

Reply:
It was 6% in my dotfiles with https://github.com/nix-community/nix-direnv/pull/225 applied, but not a very scientific benchmark. I actually have a better idea how we can efficiently watch all nix files:

```
find . -type f -name "*.nix" -newermt $date_of_our_cache
```

Even on nixpkgs this is still fast enough (with a warm cache here):

```
╭─[~/git/nixpkgs]─[main]
╰─ % time find . >/dev/null
find . > /dev/null  0,05s user 0,26s system 98% cpu 0,314 total
```

Reply:
That addresses the adding question, but we still need walk and test the modified timestamp of those files. Is *that* fast enough?

I guess I'm asking if we can have our cake and eat it too?

Reply:
I think that this is a good discussion, but I do think it is bigger than this PR initially intended to resolve. I also think we can merge this PR without COMPLETELY addressing these issues. The latest push that I made should resolve the issue with a smaller change overall. 

I'll open a new ticket talking about performance improvements (and referencing this PR, #225, and #226) and we can discuss a potential new approach there.